verification/spec/connection-handshake/chm.tla

```diff
@@ -0,0 +1,237 @@
+-------------------------------- MODULE CHM --------------------------------
+
+EXTENDS Naturals, FiniteSets
+
+
+CONSTANT MaxHeight     \* Maximum height of all the chains in the system.
+
+
+VARIABLES
+    chModule,   \* The state of this connection handshake module (chm).
+    turn,       \* Keep track of who takes a step: either chModule or environment.
+    inMsg,      \* A buffer holding any message incoming to the chModule.
+    outMsg      \* A buffer holding any message outbound from the chModule.
+
+
+vars == <<turn, chModule, inMsg, outMsg>>
+
+Heights == 1..MaxHeight
+
+ConnectionStates == {"UNINIT", "INIT", "TRYOPEN", "OPEN"}
+
+ConnectionIDs == {"connAtoB", "connBtoA"}
+ClientIDs == { "clientA", "clientB" }
+
+Null == "none"
+noMsg == [ type |-> "none" ]
+noErr == "errNone"
+
+
+(***************************** ConnectionEnds *****************************
+    A set of connection end records.
+    A connection end record contains the following fields:
+
+    - connectionID -- a connection identifier
+      Stores the connection identifier for this connection end.
+
+    - clientID -- a client identifier
+      Stores the client identifier associated with this connection end.
+
+    Optional, add later:
+    - prefix
+    
+ ***************************************************************************)
+ConnectionEnds ==
+    [
+        connectionID : ConnectionIDs \union {Null},
```

Comment:
Is there a need having Null here as connectionEnd is never initialised with null fields?